### ISSUE TITLE
The `--debug` flag has no effect (i.e., no stack trace is displayed) when the exception happens before the flag's injection

### ISSUE THREAD

**Issue by cos** [kframework/k-legacy#937]

**Comment by radumereuta** [comment]
I've encountered this a few times, it's really annoying.

**Comment by osa1** [comment]
:+1: I'm also suffering from this. Thankfully it's not very hard to modify the source temporarily to print messages all the time.

**Comment by dwightguth** [comment]
I'm not sure it's possible to fix this issue without introducing behavior that is even more undesirable than the behavior we are trying to fix. I made a stab at it, but once I had an implementation, what I realized was, if the user has a typo while typing the options, then the options won't parse, including the --debug flag, which will cause a giant nasty stack dump to appear to users even when it's a perfectly ordinary user error. Since the debug flag is intended to be of use only to K developers, I believe it is better to provide good user experience to our users by disabling the debug flag until it is explicitly enabled, and taking the hit that some developers will sometimes need to modify the code in order to generate a stack dump.

**Comment by dwightguth** [comment]
Closing this since I haven't heard any response from anyone about my answer to the issue of whether this is possible. Feel free to reopen if you can come up with a solution that does not have the issue I described.

**Comment by cos** [comment]
I fully agree with trimming down the issues and making sure someone is assigned to each. Still, this does not mean we should close down issues which have no resolution yet, or there is nobody available to fix them. I think this issue is one of them. It is annoying so we should find a solution eventually.
